### ISSUE TITLE
chore: Roll our openapi.json back into FastAPI

### ISSUE THREAD

**Issue by awalker4** [Unstructured-IO/unstructured-api#317]
The openapi.json in this repo is the source of truth for our [Speakeasy client](https://github.com/Unstructured-IO/unstructured-python-client). We want to roll these changes back into FastAPI so they get into our [hosted Swagger docs](https://api.unstructured.io/general/docs). The Speakeasy team has given us [a diff](https://github.com/ndimares/unstructured-api/commit/489f88c64274cfb2328c94de83d36f7f7b75623a#diff-c7c4c96edee21397f4d43c73169849e2f4ff05f1cfa3c5b7ea0db1456045b373) to get started, so we just need to get this merged.

Note that many of our parameters are lists when they shouldn't be. Much of the complexity here is working around that in the generated spec, ~~so it might be cleaner to just fix the params first~~. Goal for this ticket should just be to apply the linked diff. We'll address the param types elsewhere.

**Comment by cragwolfe** [comment]
> Note that many of our parameters are lists when they shouldn't be.

however, i believe at this point we need to maintain backwards compat of for pretty much all of the existing params.

**Comment by g-parki** [comment]
Hi @awalker4, I came across the "everything in the docs is a list" issue and I poked around to see if Pydantic was used or could be implemented/updated. Is there some history there? I see Pydantic listed in `base.in`, but I don't see it actually being imported (see grep below).

I'd be happy to contribute a PR for request models if that would be welcome.
```sh
grep -r "pydantic" .
./requirements/base.in:pydantic<2.0.2
./requirements/test.txt:pydantic==1.10.13
./requirements/test.txt:    #   pydantic
./requirements/base.txt:pydantic==1.10.13
./requirements/base.txt:    #   pydantic
./requirements/base.txt:    #   pydantic-core
./prepline_general/api/general.py:    # Note(austin): pydantic should control this sort of thing for us
```

**Comment by g-parki** [comment]
> I'd be happy to contribute a PR for request models if that would be welcome.

Actually I see the request is form data, which I don't think can be neatly wrapped in a Pydantic model. For backward compatibility with the array'd params, a Pydantic [annotated validator](https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators) could be used to take only the first element of the list as needed. The Swagger docs and the OpenAPI schema would only show the new types, so the backward compatibility would be undocumented.

```python
T = TypeVar("T")

def first_element_of_list(value: Union[T, list[T]]) -> T:
    if isinstance(value, list) and len(value) > 0:
        return value[0]
    return value

@router.post("/general/v0/general")
def pipeline_1(
    ...
    strategy: Annotated[
        Literal["fast", "hi_res", "auto"],
        Form(
            title="Strategy",
            description="The strategy to use for partitioning PDF/image. Options are fast, hi_res, auto. Default: auto",
            examples=["auto", "hi_res"],
        ),
        BeforeValidator(first_element_of_list),
    ] = "fast",
    ...
)
    ...
```
<img width="648" alt="image" src="https://github.com/Unstructured-IO/unstructured-api/assets/61096711/c17abf53-27fe-4afa-ae67-bc6fe3349d7e">

**Comment by awalker4** [comment]
A PR for this would be great! For context, we used to autogenerate this repo through [unstructured-api-tools](https://github.com/Unstructured-IO/unstructured-api-tools). That project generates FastAPI code from a Jupyter notebook, for quickly spinning up data pipelines. `unstructured-api` has outgrown api-tools, and many of those conventions aren't useful here.

**Comment by awalker4** [comment]
Closed with #359
